doc(python): Improve Python SDK docs / added releases and session documentation. (#4713)
getsentry/sentry-docs: src/platforms/python/common/configuration/integrations/index.mdx

---
title: Integrations
description: "Sentry provides additional integrations designed to change configuration or add instrumentation to your application."
sidebar_order: 30
---

Sentry provides a number of integrations which are designed to add automatic instrumentation to your application.

Integrations can be configured in the call to `sentry_sdk.init()` like this:

```python
import sentry_sdk
from sentry_sdk.integrations.django import DjangoIntegration

sentry_sdk.init(
    ...
    integrations=[
        DjangoIntegration(),
        CeleryIntegration(),
        RedisIntegration(),
    ],
)
```

> 

Reviewer: This appears to duplicate similar content that already exists on the child pages of this page and I don't think it needs to be repeated. Take a look at the JavaScript Integrations page for an example of what this page generally looks like. I think the prose is that you've added is fine, but the code snippet seems out of place.

Author: It is a similar content as in the child pages, but it is illustrating, that one can add multiple integrations (which is not shown elsewhere) and also it enables Django, Celery and Redis which are possibly the most used combinations for python web projects and there it can be kind of a shortcut for users doing a lot of Django projects. They just need to copy/paste this and add the dsn and they are done

Reviewer: @untitaker, your opinion on this would be helpful here. I'm concerned about this Integrations page being very different from other ones. Do you think this information is important enough to keep on this page vs (perhaps) having it as an extra code snippet on the child pages?

Author: The main thing why I put this here is that developers are lazy and don't want to think about integrating third party libraries. If your are doing a new Django project you can just go to "Configuration" -> "Integrations" then copy this snipped and you are done. 
But because we have also automatic enabling of integrations this is not necessary at all. (we do not have documented automatic enabling of integrations anywhere, but this is another discussion for another place)

So I am also fine with removing it.

Author: I have now removed the code snippet. After reading the docs guide and thinking about it, it is really a duplicated code snippet.

> **NEW:** The Sentry Python SDK also tries to automatically enable integrations that fit your Python modules you have installed. If you have Django installed the Sentry SDK will enable the `DjangoIntegration()` by default for you. No need for configuration at all!

Reviewer: Adding markers like "new" on content means you then have to think about keeping track of when it's no longer new. Err on the side of making the content future-proof. Also, always avoid exclamation points. Generally we go for an informative, but casual tone.

```suggestion
The Sentry Python SDK also tries to automatically enable integrations that fit the Python modules you have installed. If you have Django installed, the Sentry SDK will enable the `DjangoIntegration()` by default for you without any further configuration.
```

> If you do not want this behavior, set the `default_integrations` parameter to `False`. (See [Integration Configuration](/platforms/python/configuration/options/#integration-configuration) on the Basic Options page)

Reviewer: Unless there's a good reason not to, we generally use contractions. 

```suggestion
If you don't want integrations to be enabled automatically, set the `default_integrations` parameter to `False`. Learn more about <PlatformLink to="/configuration/options/#integration-configuration">integration options</PlatformLink>integration configuration.
```
